core/templates/dev/head/pages/admin-page/services/admin-data.service.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service that manages admin data.
+ */
+
+require('pages/admin-page/admin-page.constants.ts');
+
+var oppia = require('AppInit.ts').module;
+
+oppia.factory('AdminDataService', [
+  '$http', 'ADMIN_HANDLER_URL',
+  function($http, ADMIN_HANDLER_URL) {
+    var dataPromise = null;
+    var initalize = function() {
+      dataPromise = $http.get(ADMIN_HANDLER_URL).then(function(response) {
+        return response.data;
+      });
+    };
```

Comment:
I don't think we need this as a separate method since it is called just from one place.